tests/periph_spi/tests/04__periph_spi_timer.robot

```diff
@@ -0,0 +1,60 @@
+*** Settings ***
+Documentation       Verify that the meassured times are within a certain window
+
+Suite Setup         Run Keywords    PHILIP Reset
+...                                 RIOT Reset
+...                                 API Sync Shell
+...                                 API Firmware Should Match
+Test Setup          Run Keywords    PHILIP Reset
+...                                 RIOT Reset
+...                                 API Sync Shell
+...                                 SPI Init Should Succeed
+Test Teardown       Run Keywords    SPI Release Should Succeed
+
+Resource            periph_spi.keywords.txt
+Resource            api_shell.keywords.txt
+
+Variables           test_vars.py
+
+Force Tags          periph  spi
+
+
+*** Keywords ***
+SPI Clock Speed Check
+    [Documentation]     Main routine to verify the clock speed
+    [Arguments]         ${clk_speed_string}  ${clk_speed_value}  ${size}
+    API Call Should Succeed             PHiLIP.Write Reg        spi.mode.if_type    3
+    API Call Should Succeed             PHiLIP.Write Reg        spi.mode.init       0
+    API Call Should Succeed             PHiLIP.Execute Changes
+    SPI Acquire Should Succeed          0  ${clk_speed_string}
+    SPI Transfer Bytes Should Succeed   cont=0   in_len=${size}
+    API Call Should Succeed             PHiLIP.Read Reg         sys.sys_clk
+    ${sys_clk} =  Set Variable          ${RESULT['data']}
+    API Call Should Succeed             PHiLIP.Read Reg         spi.frame_ticks
```

Comment:
If frame ticks are unreliable (it is more a check of the cs pin not of the actual spi bytes) maybe you want to see `spi.byte_ticks` instead. This gets the time from the end of one byte to the end of the following one.